airflow/models/abstractoperator.py

```diff
@@ -494,18 +495,30 @@ def expand_mapped_task(self, run_id: str, *, session: Session) -> tuple[Sequence
                     total_length,
                 )
                 unmapped_ti.state = TaskInstanceState.SKIPPED
-                indexes_to_map = ()
             else:
-                # Otherwise convert this into the first mapped index, and create
-                # TaskInstance for other indexes.
-                unmapped_ti.map_index = 0
-                self.log.debug("Updated in place to become %s", unmapped_ti)
-                all_expanded_tis.append(unmapped_ti)
-                indexes_to_map = range(1, total_length)
-            state = unmapped_ti.state
-        elif not total_length:
+                zero_index_ti_exists = session.query(
+                    exists().where(
+                        TaskInstance.dag_id == self.dag_id,
+                        TaskInstance.task_id == self.task_id,
+                        TaskInstance.run_id == run_id,
+                        TaskInstance.map_index == 0,
+                    )
+                ).scalar()
```

Comment:
IIRC `EXISTS` has some compatibility issues across databases (don’t remember what exactly), so we generally use `query(count())...scalar() > 0` instead.